### PR TITLE
null cl lua emitter fix

### PR DESCRIPTION
**Pull request by meetric1**
added a NULL check for invalid cl lua emitters.

### CODE REVIEW

**Review by AmyJeanes — APPROVED** [review]
LGTM, not sure how the emitter is dying but this seems to be a very common error, the least we can do is stop it throwing exceptions for now. Will let @weltensturm have the final say on this PR though.

Also @weltensturm are you happy to either publish this to workshop or to allow me to attempt it? I believe contributors now have access to push new versions of the addon.

File: wac aircraft/lua/entities/wac_hc_base/cl_init.lua [hunk 1]
```diff
@@ -559,6 +559,7 @@ function ENT:Draw()
 	if self.engineRpm > 0.2 and self.SmokePos then
 		if !self.lastHeatDrawn or self.lastHeatDrawn < CurTime()-0.1 then
 			if type(self.SmokePos) == "table" then
+				if !self.Emitter or !self.Emitter:IsValid() then return end	-- sometimes it isnt valid for some reason
```
Comment (weltensturm) [inline wac aircraft/lua/entities/wac_hc_base/cl_init.lua:562]: I think a `and IsValid(self.Emitter) then` on line 559 would be nicer but idc

**Comment by weltensturm** [comment]
@MattJeanes feel free to try it, I don't feel well enough to touch it atm

**Comment by AmyJeanes** [comment]
@weltensturm tried it but got this, I think you may need to accept it first before it lets me:

image

Link: http://steamcommunity.com/sharedfiles/workshoplegalagreement

**Comment by weltensturm** [comment]
did that, did it do anything?

**Comment by AmyJeanes** [comment]
Yep that seems to have passed it now, thanks! All updated on workshop hopefully now will keep an eye on comments to make sure I've not broken it 😄 

Note workshop now uses delta patching which is why the file size is so small

image

**Comment by weltensturm** [comment]
thanks!

**Pull request merged**